app/views/puzzles/update.turbo_stream.erb

```diff
@@ -0,0 +1,25 @@
+<turbo-stream action="replace" target="<%= dom_id(@puzzle) %>">
+  <template>
+    <tr id="<%= dom_id(@puzzle) %>">
+      <td><%= @puzzle.question %></td>
+      <td><%= @puzzle.answer %></td>
+      <td><%= @puzzle.explanation %></td>
+      <td>
+        <% if @puzzle.link.present? %>
+          <%= link_to 'View', @puzzle.link, target: '_blank' %>
+        <% end %>
+      </td>
+      <td>
+          <%= button_to 'Approve', puzzle_state_path(@puzzle, state: :approved), method: :patch, form_class: 'inline-form', class: 'btn approve-btn' %>
+          <%= button_to 'Reject', puzzle_state_path(@puzzle, state: :rejected), method: :patch, form_class: 'inline-form', class: 'btn reject-btn' %>
+          <%= link_to "Edit",
+              edit_puzzle_path(@puzzle),
+              data: { turbo_frame: "modal" },
+              class: "btn edit-btn" %>
+      </td>
+    </tr>
```

Comment:
Maybe you can extract the row table into a partial and then reuse it here.

https://github.com/fastruby/ruby_or_rails/blob/523b1bfba58def075d29be28551a8ed27f4bf572/app/views/puzzles/_puzzles_table.html.erb#L13-L42

```suggestion
  <%= render partial: 'puzzle_raw', locals: { puzzle: @puzzle, actions: :pending } %>
```